eessi/testsuite/hooks.py

```diff
@@ -573,6 +574,15 @@ def set_compact_thread_binding(test: rfm.RegressionTest):
     log(f'Set environment variable KMP_AFFINITY to {test.env_vars["KMP_AFFINITY"]}')
 
 
+def set_omp_num_threads(test: rfm.RegressionTest):
+    """
+    Set default number of OpenMP threads equal to number of CPUs per task,
+    unless OMP_NUM_THREADS is already set
+    """
+    test.env_vars['OMP_NUM_THREADS'] = os.getenv('OMP_NUM_THREADS', test.num_cpus_per_task)
```

Comment:
I don't think it makes sense to do this conditionally. What you are checking is if `OMP_NUM_THREADS` is set on the node where the `reframe` command runs (typically: login node). What you are setting is an `export OMP_NUM_THREADS=<something>` in the job script, which will cause it to be set on the batch node running the job.

Those are typically environments that have nothing to do with each other: we advise (in our docs) to configure reframe to submit with `--export=None` (https://www.eessi.io/docs/test-suite/ReFrame-configuration-file/) , precisely to make sure the submission environment doesn't affect the environment of the batch job.

I'd suggest just setting this in the test as `test.env_vars['OMP_NUM_THREADS'] = test.num_cpus_per_task`, unconditionally (pretty much similar to what we do for GROMACS, even if that is a bit more complex: https://github.com/EESSI/test-suite/blob/926c3e4859f2c9dbe0b9034202d317cdce331fba/eessi/testsuite/tests/apps/gromacs.py#L99)